File: api/schemas/pets.py
from pydantic import BaseModel
from typing import Optional


class PetCreate(BaseModel):
    nome: str
    especie: str
    raca: str
    dataNascimento: str
    idTutor: int


class PetUpdate(BaseModel):
    nome: Optional[str] = None
    especie: Optional[str] = None
    raca: Optional[str] = None
    dataNascimento: Optional[str] = None
    idTutor: Optional[int] = None


class PetResponse(BaseModel):
    idPet: int
    nome: str
    especie: str
    raca: str
    dataNascimento: str
    idTutor: int

File: frontend/src/services/pets.ts
import { api } from '@/lib/api'
import type { Pet } from '@/types'

interface MessageResponse { mensagem: string; sucesso: boolean }

export interface PetCreate {
  nome: string
  especie: string
  raca: string
  dataNascimento: string
  idTutor: number
}

export const petsService = {
  listar:    ()                               => api.get<Pet[]>('/pets/'),
  cadastrar: (body: PetCreate)                => api.post<MessageResponse>('/pets/', body),
  atualizar: (id: number, body: Partial<PetCreate>) => api.put<MessageResponse>(`/pets/${id}`, body),
}
